app/src/main/java/org/mozilla/fenix/components/metrics/UsageRecorder.kt

```diff
@@ -0,0 +1,44 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.components.metrics
+
+import android.app.Activity
+import android.content.Context
+import mozilla.components.support.utils.ext.getPackageInfoCompat
+import org.mozilla.fenix.android.DefaultActivityLifecycleCallbacks
+import org.mozilla.fenix.utils.Settings
+
+/**
+ * This will store app usage time to disk, based on Resume and Pause lifecycle events. Currently,
+ * there is only interest in usage during the first day after install.
+ */
+class UsageRecorder(private val settings: Settings) : DefaultActivityLifecycleCallbacks {
+    private val activityStartTimes: MutableMap<String, Long?> = mutableMapOf()
+    private val dayMillis: Long = 1000 * 60 * 60 * 24
+
+    override fun onActivityResumed(activity: Activity) {
+        super.onActivityResumed(activity)
+        activityStartTimes[activity.componentName.toString()] = System.currentTimeMillis()
+    }
+
+    override fun onActivityPaused(activity: Activity) {
+        super.onActivityPaused(activity)
+        val startTime = activityStartTimes[activity.componentName.toString()] ?: return
+        val elapsedTimeMillis = System.currentTimeMillis() - startTime
+        settings.usageTimeGrowthData = elapsedTimeMillis + settings.usageTimeGrowthData
```

Comment:
I don't expect this to be foolproof, but concerns have been raised to stakeholders about the unreliability of using lifecycle events to achieve this. It seems like this type of solution is good enough from their perspective, but I am open to any suggestions on how we might better achieve this.

Reply:
This is fine, but doesn't give a clear understanding of what is happening in the activity. Are they in onboarding a custom tab (yes, you can use Firefox as a custom tab for your first session without opening the main app), browsing?

> I don't expect this to be foolproof, but concerns have been raised to stakeholders about the unreliability of using lifecycle events to measure this.

`ActivityLifecycleCallbacks` is the best option I can think of as well.

Reply:
@jonalmeida I'm not sure I fully understand your concern here. We want to measure usage across _any_ activities (even the case you provided) so I don't think we need to be concerned with what is actually happening in the activity. Is that what I could make more clear in the class documentation perhaps?

Reply:
Sorry, I just meant the data might not be super useful to the final party. It's more a note than a concern. Regardless, you're taking the right direction by using `ActivityLifecycleCallbacks`.